### PR TITLE
discovery: Consolidate procfs access in system-probe

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

Currently, procfs access is done both from the agent check and from system-probe. The are a couple of issues with the procfs access in the agent check:
- It does not access `HOST_PROC` since it uses prometheus' procfs with `NewDefaultFS`, unlike the system-probe which either uses gopsutil (which respects `HOST_PROC` internally) or explicitly uses the host proc using the appropriate helper functions.
- It duplicates the work done by system-probe. For example, the list of all running processes is constructed twice, once in the agent check, and once in system-probe

This consolidates the procfs access in system-probe and removes it from the agent check and will allow further optimizations and cleanups.

### Motivation

Noticed while working on https://github.com/DataDog/datadog-agent/pull/28931

### CODE REVIEW

**Review by guyarb — COMMENTED** [review]
Why not moving core-check logic into system-probe?

File: pkg/collector/corechecks/servicediscovery/module/impl_linux.go [hunk 1]
```diff
@@ -219,6 +226,11 @@ func (s *discovery) getServiceInfo(proc *process.Process) (*serviceInfo, error)
 		return nil, err
 	}
 
+	createTime, err := proc.CreateTime()
+	if err != nil {
+		return nil, err
+	}
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/module/impl_linux.go:232]: That is out of scope for the PR as the gap exists from day 1, but getting the creation time of a process without timezone information, is going to be problematic for customers
I think we need to re-work the creation time and gain all information we need to properly show it to customers

cc @LotharSee

File: pkg/collector/corechecks/servicediscovery/impl_linux.go [hunk 2]
```diff
@@ -139,23 +107,7 @@ func (li *linuxImpl) DiscoverServices() (*discoveredServices, error) {
...
-				telemetryFromError(errWithCode{
-					err:  err,
-					code: errorCodeProcfs,
-					svc:  nil,
-				})
-				log.Errorf("[pid: %d] failed to get process info: %v", pid, err)
-				li.ignoreProcs[pid] = true
-				continue
-			}
+			svc := li.getServiceInfo(pid, service)
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/impl_linux.go:110]: `svc` is a pointer, to be future-proof - please check it is not nil before accessing to its attributes 

Also, why do we return a pointer instead of a value?

**Review by guyarb — APPROVED** [review]

File: pkg/collector/corechecks/servicediscovery/impl_linux.go (hunk 2 above)
Comment (vitkyrka) [reply pkg/collector/corechecks/servicediscovery/impl_linux.go:110]: Done, removed pointer return.
